### PR TITLE
feat: auth added

### LINKED ISSUE #11: Add authentication
- Add users to users database.
- Connect auth forms with backend.
- Make signup and login live.

### PR DESCRIPTION
**Pull request by rdshinde**
- signup feat added
- login feat added
- password component for matching passwords also added.

chrome-capture-2022-2-24

### CODE REVIEW

**Review by rahulcs754 — APPROVED** [review]
looks all good and clean code.
nice design

File: src/components/authentication-form/LoginForm.jsx [hunk 1]
```diff
@@ -1,18 +1,97 @@
...
+    setApiUrl("/api/auth/login");
+  };
+  const { isLoaderLoading, serverResponse } = useAxios(apiURL, "POST", {
+    email: email,
+    password: password,
+  });
+  useEffect(() => {
+    const { data, status } = serverResponse;
+    if (status === 200) {
+      localStorage.setItem("token", JSON.stringify(data.encodedToken));
```
Comment (kjais1720) [inline src/components/authentication-form/LoginForm.jsx]: You don't need to stringify the token here, as its already a string

File: src/components/authentication-form/LoginForm.jsx [hunk 2]
```diff
@@ -1,18 +1,97 @@
...
+      userAuthDispatch({
+        type: "LOGIN",
+        payload: {
+          encodedToken: data.encodedToken,
+          user: { ...data.foundUser },
+        },
+      });
+      navigate("/");
+    }
+    console.log("sideeffect");
```
Comment (kjais1720) [inline src/components/authentication-form/LoginForm.jsx]: remove the console.log

File: src/components/authentication-form/LoginForm.jsx (hunk 1 above)
Comment (rdshinde) [reply src/components/authentication-form/LoginForm.jsx]: Yes got it thanks for pointing out.

File: src/components/authentication-form/LoginForm.jsx (hunk 2 above)
Comment (rdshinde) [reply src/components/authentication-form/LoginForm.jsx]: Ok, thanks for pointing out.

**Review by kjais1720 — APPROVED** [review]
All good now 👍

**Pull request merged**